roles/set_custom_fact/tasks/main.yml

```diff
@@ -145,3 +129,14 @@
 - name: Set variable generic_project_name
   set_fact: generic_project_name={{ ansible_hostname | regex_replace('-\d+', '') }}
   tags: [ 'master', 'custom', 'rename', 'update','idc_import','package_management','kalite_import' ]
+
+- name: Stat /dev/sda1
+  stat: path=/dev/sda1
+  register: sda1
+  tags: ['always']
+
+- name: Use sda1 as external hard drive if sda1 exists and on arm architecture
+  set_fact: ext_hdd={{ (ansible_architecture == "armhf" or ansible_architecture == "armv7l") and (sda1.stat.isblk | default (omit)) == True }}
+  when: firststart.stat.exists == false or firststart is undefined
```

Comment:
If you use this condition, this task will be executed only when `/etc/firststart.csv` does not exist, meaning when the device has no device name defined. However `ext_hdd` is going to be used through the whole playbook, this will lead to an error. 

Either you store the `ext_hdd` variable in custom fact in order to be accessible later on in the playbook or you remove the condition.